app/models/cherrypick/volume_by_nano_grams_per_micro_litre.rb

```diff
@@ -5,70 +5,82 @@ module Cherrypick::VolumeByNanoGramsPerMicroLitre
   # Used in the cherrypicking process to calculate the relative volumes of source
   # and buffer required to reach the target volume and concentration.
   #
-  # - Ideally will aim for a target well containing volume_required at concentration_required
-  # - This will be made by combining material from the source well (picked_volume) and
-  #   buffer (buffer_volume)
-  # - Maximum picked_volume is limited by source_volume and volume_required
-  # - If source_volume < volume_required then buffer will be added to make up volume_required
-  #   even if it reduces the target concentration below concentration_required
+  # - Ideally will aim for a target well containing volume `final_volume_desired`
+  #   at concentration `final_conc_desired`
+  # - This will be made by combining material from the source well (well_attribute.picked_volume)
+  #   and buffer (well_attribute.buffer_volume)
+  # - Maximum picked_volume is limited by the amount available (source_volume) and the amount you want
+  #   in the target well (final_volume_desired)
+  # - If source_volume < final_volume_desired then buffer will be added to make up final_volume_desired
+  #   even if it reduces the target concentration below final_conc_desired
   #
-  # @param volume_required [Float] The volume to aim for in the target well (self)
-  # @param concentration_required [Float] The concentration to aim for in the target well (self)
+  # @param final_volume_desired [Float] The volume to aim for in the target well (self)
+  # @param final_conc_desired [Float] The concentration to aim for in the target well (self)
   # @param source_concentration [Float] The concentration (ng/ul) in the source well
   # @param source_volume [Float] The volume (ul) in the source well (ie. the maximum pick)
-  # @param robot_minimum_picking_volume [Float] (ul) the minimum volume the robot can pick
+  # @param robot_minimum_pick_vol [Float] (ul) the minimum volume the robot can pick (the robot
+  #        software will reject requests to pick less than this)
   #
-  # @return [Float] The total volume that will be picked from the source well
+  # @return [Float] The volume that we will instruct to robot to pick from the source well
   #
   # rubocop:todo Metrics/MethodLength, Metrics/AbcSize
   def volume_to_cherrypick_by_nano_grams_per_micro_litre(
-    volume_required,
-    concentration_required,
+    final_volume_desired,
+    final_conc_desired,
     source_concentration,
     source_volume,
-    robot_minimum_picking_volume = 0.0
+    robot_minimum_pick_vol = 0.0
   )
-    robot_minimum_picking_volume ||= 0
+    robot_minimum_pick_vol ||= 0
```

Comment:
Should this not be set to a float rather than an integer?  I'm sure it doesn't make much difference, but it's weird to see this setting to `0` when the default for the argument above is `0.0`.

Reply:
I didn't notice this. I'm sure Ruby does all sorts of magical conversions. Is this line actually redundant because of the default in the parameter list?

Reply:
Ruby does distinguish between integers and floats.  If you do a division with two integers, you get integer results only.  `3/2 = 1`

I'm not sure about the redundancy.  If someone forcefully passes `nil` as the value for `robot_minimum_pick_vol` does it use the default value or is that only when you don't pass any value at all?

Reply:
You're right, it's not redundant - there are a few tests that specifically pass in `nil`.
I'm going to go with "if it ain't broke, don't fix it" :)

Reply:
I know you've merged, but as a rule of thumb I would say that if you see something that you're fairly certain isn't right and when you fix it the tests still pass, change it.  It's subtle things like this that maybe aren't causing problems today, but down the line someone makes a change where there is a difference between having a float and having and integer and then the bugs can be quite hard to identify.